### DISCUSSION TITLE
Cobbler 2.8 to deploy Ubuntu 20.04 (or 22.04)

### DISCUSSION THREAD

**Discussion by sysadmincaos** [cobbler/cobbler#3550] [Q&A]
Hello,

my provisioning server is a Cobbler 2.8.0-4 running in an CentOS-7.9.2009. Yes, I know this server is oooold, but works fine for what I need it.
Always I have been deploying RedHad based distributions. Some years ago, CentOS-7 and, now, Rocky-8. However, now I want to try to deploy a Debian based distribution like Ubuntu. I have tried Ubuntu Server Live 20.04.6 and Ubuntu Server Legacy 20.04.1. I have added a small kickstart file "ks" with this configuration (only for testing purposes):

```
lang en_US.UTF-8
keyboard us
text
skipx
user --name=name --password=psswd
reboot
url --url=http://192.168.60.10/cblr/links/Ubuntu-20_04-casper-x86_64/
network --bootproto=static --ip=192.168.11.106 --netmask=255.255.255.0 --gateway=192.168.11.253 --nameserver=158.109.0.1,8.8.8.8 --device=enp0s25 --hostname=nodo1-ngl --onboot=on --noipv6 --activate
rootpw --plaintext passwd
firewall --disabled
authselect --enableshadow --passalgo=sha512
selinux --permissive
timezone --utc Europe/Madrid
zerombr
clearpart --all --drives=sda --initlabel
bootloader --location=mbr --driveorder=sda
autopart --type=lvm
firstboot --enable

%packages
ubuntu-minimal
openssh-server
screen
curl
wget
acpid
unattended-upgrades
linux-image-generic
python-apt
lshw
lldpd
dmidecode
%end
```

Also, I have created a seed file with this content... but, honestly, I don't know if I need the "seed" file or only with "ks" I could deploy a new system. This "seed" file is this:
```
d-i     debian-installer/locale string es_ES.UTF-8
d-i     debian-installer/splash boolean false
d-i     console-setup/ask_detect        boolean false
d-i     console-setup/layoutcode        string us
d-i     console-setup/variantcode       string
d-i     netcfg/get_nameservers  string 158.109.0.49
d-i     netcfg/get_ipaddress    string 192.168.11.150
d-i     netcfg/get_netmask      string 255.255.255.0
d-i     netcfg/get_gateway      string 192.168.11.253
d-i     netcfg/confirm_static   boolean true
d-i     clock-setup/utc boolean true
d-i     partman-auto/method string regular
d-i     partman-lvm/device_remove_lvm boolean true
d-i     partman-lvm/confirm boolean true
d-i     partman/confirm_write_new_label boolean true
d-i     partman/choose_partition        select Finish partitioning and write changes to disk
d-i     partman/confirm boolean true
d-i     partman/confirm_nooverwrite boolean true
d-i     partman/default_filesystem string ext3
d-i     clock-setup/utc boolean true
d-i     clock-setup/ntp boolean true
d-i     clock-setup/ntp-server  string ntp.ubuntu.com
d-i     base-installer/kernel/image     string linux-server
d-i     passwd/root-login       boolean false
d-i     passwd/make-user        boolean true
d-i     passwd/user-fullname    string ubuntu
d-i     passwd/username string ubuntu
d-i     passwd/user-password-crypted    password $6$gWaHgdmxD/uiQIPh$mJoTZ/d33KM9P64PSwP9FLT6EgmZCnpAjuKj8GQGSAaTBog.FVD7QFqaSn5lIlnK.ad7LRX.lkVE7U6wyPN.U0
d-i     passwd/user-uid string
d-i     user-setup/allow-password-weak  boolean false
d-i     user-setup/encrypt-home boolean false
d-i     passwd/user-default-groups      string adm cdrom dialout lpadmin plugdev sambashare
d-i     apt-setup/services-select       multiselect security
d-i     apt-setup/security_host string security.ubuntu.com
d-i     apt-setup/security_path string /ubuntu
d-i     debian-installer/allow_unauthenticated  string false
d-i     pkgsel/upgrade  select safe-upgrade
d-i     pkgsel/language-packs   multiselect
d-i     pkgsel/update-policy    select none
d-i     pkgsel/updatedb boolean true
d-i     grub-installer/skip     boolean false
d-i     lilo-installer/skip     boolean false
d-i     grub-installer/only_debian      boolean true
d-i     grub-installer/with_other_os    boolean true
d-i     finish-install/keep-consoles    boolean false
d-i     finish-install/reboot_in_progress       note
d-i     cdrom-detect/eject      boolean true
d-i     debian-installer/exit/halt      boolean false
d-i     debian-installer/exit/poweroff  boolean false
d-i     pkgsel/include string byobu vim openssh-server
byobu   byobu/launch-by-default boolean true
```

Then, I have imported ISO file and reconfigured distro in Cobbler to assign my new "ks" file.

However, when I start a new deployment, in both Ubuntu distros, installation "stops" when it tries to detect CD-ROM... 

What can I do? What am I doing wrong?

Thanks!!!

**Comment by SchoolGuy** [comment]
I sadly have no dedicated knowledge about 2.x.x versions of Cobbler. However, the principles are the same as with Cobbler 3.x.x. As such the issue most likely is indeed with the template as you mentioned.

https://github.com/cobbler/cobbler/blob/8ca9d90761ad6bb5f0a4342c7db43a08c8f2d500/autoinstall_templates/sample.seed#L34-L38

I personally sadly havn't had the time to fully auto-install a Debian-based OS so far but I figure that something along these lines is missing in your preseed file.

**Comment by btravouillon** [comment]
Kickstart files are not for Debian-based distro which use pre-seed files.

The Ubuntu 20.04 legacy installer will support a pre-seed file. However, this installer has been deprecated with 20.04 and removed in 22.04. Ubuntu now uses [Autoinstall](https://ubuntu.com/server/docs/install/autoinstall). This installer is not yet fully supported by Cobbler. You may find further information in https://github.com/cobbler/cobbler/discussions/3217.

Anyway, I would definitely recommend you to upgrade your Cobbler install sooner than later.
